### ISSUE TITLE
Tests failing blocking install, offline tests attempt to touch real user credentials during install!

### ISSUE THREAD

**Issue by MattOates** [zostay/Amazon-DynamoDB#1]
As part of the unbitrot hackathon I came across the following. Im not sure you necessarily want to touch a users real credentials file from `~.aws/credentials` during install. But either way, trying to touch it blocks the install if the AWS credentials file doesn't exist for the user.

> $ zef install Amazon::DynamoDB
> ===> Searching for: Amazon::DynamoDB
> ===> Updated cpan mirror: https://raw.githubusercontent.com/ugexe/Perl6-ecosystems/master/cpan1.json
> ===> Updated p6c mirror: http://ecosystem-api.p6c.org/projects1.json
> ===> Searching for missing dependencies: AWS::Session, WebService::AWS::Auth::V4, String::CRC32
> ===> Searching for missing dependencies: Config::INI, DateTime::Format::W3CDTF
> ===> Testing: Config::INI
> ===> Testing [OK] for Config::INI
> ===> Testing: DateTime::Format::W3CDTF
> ===> Testing [OK] for DateTime::Format::W3CDTF
> ===> Testing: AWS::Session:ver<0.4>:auth<github:zostay>
> ===> Testing [OK] for AWS::Session:ver<0.4>:auth<github:zostay>
> ===> Testing: WebService::AWS::Auth::V4:ver<0.0.3>:auth<Brad Clawsie>
> ===> Testing [OK] for WebService::AWS::Auth::V4:ver<0.0.3>:auth<Brad Clawsie>
> ===> Testing: String::CRC32:ver<0.05>:auth<cosimo>
> ===> Testing [OK] for String::CRC32:ver<0.05>:auth<cosimo>
> ===> Testing: Amazon::DynamoDB:ver<0.1>:auth<github:zostay>
> Failed to open file /Users/matt/.aws/credentials: No such file or directory
>   in sub parse_file at /Users/matt/.zef/store/perl6-Config-INI.git/4e5494798dd0f8e73045cdb36d91302ba666410e/lib/Config/INI.pm (Config::INI) line 41
>   in method get-credentials at /Users/matt/.zef/store/AWS-Session-0.4.tar.gz/AWS-Session-0.4/lib/AWS/Session.pm6 (AWS::Session) line 243
>   in method load at /Users/matt/.zef/store/AWS-Session-0.4.tar.gz/AWS-Session-0.4/lib/AWS/Credentials.pm6 (AWS::Credentials) line 212
>   in method load at /Users/matt/.zef/store/AWS-Session-0.4.tar.gz/AWS-Session-0.4/lib/AWS/Credentials.pm6 (AWS::Credentials) line 248
>   in sub load-credentials at /Users/matt/.zef/store/AWS-Session-0.4.tar.gz/AWS-Session-0.4/lib/AWS/Credentials.pm6 (AWS::Credentials) line 265
>   in method credentials at /Users/matt/.zef/store/Amazon-DynamoDB-0.1.tar.gz/Amazon-DynamoDB-0.1/lib/Amazon/DynamoDB.pm6 (Amazon::DynamoDB) line 756
>   in method access-key at /Users/matt/.zef/store/Amazon-DynamoDB-0.1.tar.gz/Amazon-DynamoDB-0.1/lib/Amazon/DynamoDB.pm6 (Amazon::DynamoDB) line 760
>   in method make-ddb-request at /Users/matt/.zef/store/Amazon-DynamoDB-0.1.tar.gz/Amazon-DynamoDB-0.1/lib/Amazon/DynamoDB.pm6 (Amazon::DynamoDB) line 785
>   in method BatchGetItem at /Users/matt/.zef/store/Amazon-DynamoDB-0.1.tar.gz/Amazon-DynamoDB-0.1/lib/Amazon/DynamoDB.pm6 (Amazon::DynamoDB) line 847
>   in block <unit> at t/aws-examples-offline.t line 57
> 
> ===> Testing [FAIL]: Amazon::DynamoDB:ver<0.1>:auth<github:zostay>
> Aborting due to test failure: Amazon::DynamoDB:ver<0.1>:auth<github:zostay> (use --force-test to override)

**Comment by zostay** [comment]
Eek. That's not good.

**Comment by zostay** [comment]
Okay, so a quick investigation shows a couple problems:

1. There are 3 tests. Of these, 2 will connect to a DDB instance using available credentials, but these must be activated with env, so they don't actually use credentials present unless told to. The third test does not connect to AWS and this is the one that is dying. It loads the default AWS::Credentials object as it would in normal operation. However, it shouldn't.
2. I am pretty sure this reveals a bug in AWS-Session, though. As I've always got a credentials file on the machines I've tested this on, I've never ran across it. I filed zostay/AWS-Session#2 to look into this problem.

For this project, I will update the code in [Test::Amazon::DynamoDB](https://github.com/zostay/Amazon-DynamoDB/blob/master/t/lib/Test/Amazon/DynamoDB.pm6) to make sure the default DDB connector creator also replaces AWS::Credentials with custom code that only concerns itself with the available environment.
